databuilder/databuilder/publisher/neo4j_csv_unwind_publisher.py

```diff
@@ -0,0 +1,471 @@
+    """
+
+    def init(self, conf: ConfigTree) -> None:
+        conf = conf.with_fallback(DEFAULT_CONFIG)
+
+        self._count: int = 0
+        self._node_files = self._list_files(conf, NODE_FILES_DIR)
+        self._node_files_iter = iter(self._node_files)
+
+        self._relation_files = self._list_files(conf, RELATION_FILES_DIR)
+        self._relation_files_iter = iter(self._relation_files)
+
+        self._driver = self._driver_init(conf)
+        self._db_name = conf.get_string(NEO4J_DATABASE_NAME)
+        self._transaction_size = conf.get_int(NEO4J_TRANSACTION_SIZE)
+
+        # config is list of node label.
+        # When set, this list specifies a list of nodes that shouldn't be updated, if exists
+        self._create_only_nodes = set(conf.get_list(NEO4J_CREATE_ONLY_NODES, default=[]))
+        self._labels: Set[str] = set()
+        self._publish_tag: str = conf.get_string(JOB_PUBLISH_TAG)
+        self._additional_publisher_metadata_fields: Dict = conf.get(ADDITIONAL_PUBLISHER_METADATA_FIELDS)
+        self._add_publisher_metadata: bool = conf.get_bool(ADD_PUBLISHER_METADATA)
+        self._publish_reverse_relationships: bool = conf.get_bool(PUBLISH_REVERSE_RELATIONSHIPS)
+        self._preserve_adhoc_ui_data = conf.get_bool(PRESERVE_ADHOC_UI_DATA)
+        if self._add_publisher_metadata and not self._publish_tag:
+            raise Exception(f'{JOB_PUBLISH_TAG} should not be empty')
+
+        LOGGER.info('Publishing Node csv files %s, and Relation CSV files %s',
+                    self._node_files,
+                    self._relation_files)
+
+    def _driver_init(self, conf: ConfigTree) -> Neo4jDriver:
+        uri = conf.get_string(NEO4J_END_POINT_KEY)
+        driver_args = {
+            'uri': uri,
+            'max_connection_lifetime': conf.get_int(NEO4J_MAX_CONN_LIFE_TIME_SEC),
+            'auth': (conf.get_string(NEO4J_USER), conf.get_string(NEO4J_PASSWORD)),
+        }
+
+        # if URI scheme not secure set `trust`` and `encrypted` to default values
+        # https://neo4j.com/docs/api/python-driver/current/api.html#uri
+        _, security_type, _ = parse_neo4j_uri(uri=uri)
+        if security_type not in [SECURITY_TYPE_SELF_SIGNED_CERTIFICATE, SECURITY_TYPE_SECURE]:
+            default_security_conf = {'trust': neo4j.TRUST_ALL_CERTIFICATES, 'encrypted': True}
+            driver_args.update(default_security_conf)
+
+        # if NEO4J_VALIDATE_SSL or NEO4J_ENCRYPTED are set in config pass them to the driver
+        validate_ssl_conf = conf.get(NEO4J_VALIDATE_SSL, None)
+        encrypted_conf = conf.get(NEO4J_ENCRYPTED, None)
+        if validate_ssl_conf is not None:
+            driver_args['trust'] = neo4j.TRUST_SYSTEM_CA_SIGNED_CERTIFICATES if validate_ssl_conf \
+                else neo4j.TRUST_ALL_CERTIFICATES
+        if encrypted_conf is not None:
+            driver_args['encrypted'] = encrypted_conf
+
+        return GraphDatabase.driver(**driver_args)
```

Comment:
add the connectivity check here: https://docs.google.com/presentation/d/109rwL8_8AyRaMa2a_nyffmfRGMusVRhZSZUymFDHw3Y/mobilepresent?slide=id.g12e46523937_0_299